src/store/RestaurantsContext.jsx

```diff
@@ -0,0 +1,51 @@
+import { useState, createContext, useEffect } from 'react';
+import { fetchRestaurants } from '../api/restaurants';
+
+export const RestaurantsContext = createContext();
+
+export const RestaurantsProvider = ({ children }) => {
+  const [restaurants, setRestaurants] = useState([]);
+
+  useEffect(() => {
+    fetchRestaurants().then((res) => setRestaurants(res));
+  }, []);
+
+  const restaurantsFilterHandler = async (filter) => {
+    const data = await fetchRestaurants();
+
+    if (filter == 'all') {
+      setRestaurants(data);
+    } else if (filter == 'popular') {
+      const popularRestaurants = data.filter(
+        (restaurant) => restaurant.rating >= 4
+      );
+
+      setRestaurants(popularRestaurants);
+    } else if (filter == 'trend') {
+      const trendingRestaurants = data.filter(
+        (restaurant) => restaurant.trending
+      );
+
+      setRestaurants(trendingRestaurants);
+    } else if (filter == 'latest') {
+      const compareTime = new Date('2023-08-23T14:15:00');
```

Comment:
Guess this is for current date comparison testing.

Reply:
You're right.